### ISSUE TITLE
Typescript: Property 'hls' does not exist on type 'Tech'

### ISSUE THREAD

**Issue by Barteque** [videojs/http-streaming#796]
## Description
I have a code that looks like that:
```
const player: VideoJsPlayer = this.getPlayer();
const playerTech: videojs.Tech = player.tech({IWillNotUseThisInPlugins: true});

const playlist = playerTech.hls.playlists.media();
```
Typescript compiler says: 
`Property 'hls' does not exist on type 'Tech'`

What can I do?

### videojs-http-streaming version
videojs-http-streaming@2.11.0

### videojs version
video.js@7.7.5

### @types/video.js version
@types/video.js@7.3.5

### Platforms
React-native-web

**Comment by gkatsev** [comment]
We don't provide types for VHS and it seems like there aren't any types in DefinitelyTyped for VHS.
I think you can cast it as `any` or something to get it to work? I'm don't really know typescript.

@joeflateau is there a quick answer here beyond adding types of VHS in DT? I don't think I saw VHS in DT.

**Comment by joeflateau** [comment]
@gkatsev since VHS is built into videojs now, I would say that the simplest approach would be to add the VHS types right into @types/videojs unless you have some argument against it. I don't see any reason to go back and make types available for use w/ v6.

**Comment by gkatsev** [comment]
My main concern is around the fact that the property isn't always available, for example on Safari or if you use a version of Video.js sans-VHS. In addition, the fact that all our docs refer to `.hls` even when usage for dash is less than ideal.

**Comment by joeflateau** [comment]
The property not always being available is not unique to VHS. We'd just need to mark the `hls` property of `videojs.Player` as optional to indicate it may or may not actually be present. You mention version of video.js sans-VHS... if you mean versions before 7, then that's not a problem because the types are set to version 7 now, if you mean a custom build of video.js w/o VHS included, then that's still fine because at that point, the user is no longer using "video.js" and the types no longer apply.

**Comment by joeflateau** [comment]
What would make me say to do it as a separate DT project is if you had plans to un-bundle VHS in the near future, but I don't believe that's the case. If you do decide somewhere down the road that it should be unbundled, then we can extract the VHS types from @types/videojs and into a separate package at that time.

**Comment by gkatsev** [comment]
Yeah, sounds like we should add it in but I'd want to hold off for a bit while we straighten out our internal usage of the `.hls`/`.dash`/`.vhs` properties and also once we ship the master branch of ths project (hopefully soon) just so that we won't need to update it again once it's shipped.

**Comment by joeflateau** [comment]
sounds good, ping me when that happens?

**Comment by gkatsev** [comment]
Will do, appreciate your help on this @joeflateau.

**Comment by gkatsev** [comment]
Once we straighten out our usage of the props would probably be a good time to start if you're working off of the master branch but we'll make sure to notify here when that happens.

**Comment by gkatsev** [comment]
That PR has been merged and VHS 2.1 is in pre-release and will likely to become latest this week!

**Comment by joeflateau** [comment]
I will take a look next week

**Comment by gkatsev** [comment]
awesome, thank you!

**Comment by Barteque** [comment]
But why video.js package still uses "1.13.3"

**Comment by gkatsev** [comment]
VHS 2.1 is in pre-release. It will be promoted `latest` this week. When it is promoted to `latest`, it'll be added to Video.js itself.

**Comment by Barteque** [comment]
This is good news :-)

**Comment by popler** [comment]
Hi, any progress to add VHS 2.1 to video.js?

**Comment by gkatsev** [comment]
we ran into a couple of bugs that have been fixed this week. We have to have another release of VHS tomorrow and then in Video.js soon.

**Comment by popler** [comment]
Thank you

**Comment by popler** [comment]
I hope it'll fix our issue with unsync audio video with alternative audio trakcs (live HLS). I tried to use VHS 2.1 with current videojs and there is no unsync problem, but we have a lot JS error and HLS levels are not working.

**Comment by gkatsev** [comment]
How does your stream work on https://videojs-http-streaming.netlify.app/? If there're still errors and what not, if you can open a new issue with an example stream, we can take a look.

**Comment by popler** [comment]
It looks better. May I have access to js file with this videojs version?

**Comment by gkatsev** [comment]
That page has whatever is in the `main` branch deployed. We're going to be making a release with that soon.

**Comment by popler** [comment]
Ok, I'm waiting. We have to use old flowplayer with hlsjs for multiaudio :(

**Comment by popler** [comment]
I saw, that you published new version of VHS (2.2.0). Do you have ETA for videojs with this http-streaming version?

**Comment by gkatsev** [comment]
We hope to get it out this week but I can't make any promises.
